integration-tests/integration_test.go

```diff
@@ -802,3 +802,81 @@ func (suite *ApexIntegrationSuite) TestRelayNAT() {
 	require.NoError(err)
 	assert.Equal(5, lc, "the number of expected wg show peers was %d, found %d: wg show out: \n%s", 5, lc, wgSpokeShow)
 }
+
+func (suite *ApexIntegrationSuite) TestApexCtl() {
+	assert := suite.Assert()
+	require := suite.Require()
+	parentCtx := context.Background()
+	ctx, cancel := context.WithTimeout(parentCtx, 60*time.Second)
+	defer cancel()
+
+	token, err := getToken(ctx, "kitteh5@apex.local", "floofykittens")
+	require.NoError(err)
+
+	// create the nodes
+	node1 := suite.CreateNode(ctx, "node1", []string{defaultNetwork})
+	suite.T().Cleanup(func() {
+		if err := node1.Terminate(parentCtx); err != nil {
+			suite.logger.Errorf("failed to terminate container %v", err)
+		}
+	})
+	node2 := suite.CreateNode(ctx, "node2", []string{defaultNetwork})
+	suite.T().Cleanup(func() {
+		if err := node2.Terminate(parentCtx); err != nil {
+			suite.logger.Errorf("failed to terminate container %v", err)
+		}
+	})
+
+	c, err := newClient(ctx, token)
+	require.NoError(err)
+
+	// create a new zone
+	zoneID, err := c.CreateZone("apexctl-zone",
+		"zone full of apexctls",
+		"172.19.100.0/24",
+		false)
+	require.NoError(err)
+
+	// TODO: debug the non-utf8 character coming back from suite.containerExec --> �{
```

Comment:
I am pretty sure `suite.containerExec()` is prepending some non-uff8 garbage on the front of the return here. 

```
�{
            "id": "8286a897-33a5-46d5-b0e8-d990c59b0eb2",
            "name": "zone-apexctl",
``` 

Getting it with all outputs from the ctl.

Reply:
Here are some example operations and outputs from the PR.


### Zone List ###

```
apexctl  --username=kitteh1@apex.local --password=floofykittens zone list
ZONE ID                                  NAME                  CIDR                DESCRIPTION                       RELAY/HUB ENABLED
0bbc1040-f1e4-4dfb-8f0c-4f7a3caf1f2c     apexctl-zone          172.19.100.0/24     zone full of apexctls             false
4e8f5374-9f73-41aa-b1fd-dfc7ae7cbc5f     zone-child-prefix     100.64.100.0/24     zone full of toddler prefixes     false
1983589e-e5d9-4de0-8410-a05300dceef2     zone-relay            10.162.0.0/24       zone with a relay hub             true
89e7abd1-4627-40f8-9daf-6fcd41ff39c9     default               10.200.1.0/20       Default Zone                      false
2978ce28-287b-49fe-af1c-c455f802ee4b     zone-blue             10.140.0.0/24       zone full of blue things          false

```

### Patch User Into Zone ###

```
apexctl  --username=kitteh1@apex.local --password=floofykittens zone move-user --zone-id 0b087540-2b06-4206-a413-e7061e81827c
```

### List Peers in a Zone

```
apexctl  --username=kitteh1@apex.local --password=floofykittens peer list --zone-id 32956ffd-3a6a-472e-82ad-a40b49b13b95
PEER ID                                  NODE ADDRESS     ENDPOINT IP             REFLEXIVE IPv4     ALLOWED IPS         RELAY NODE     CHILD PREFIX
9a755246-dd41-4175-80a5-fb4179d7cc70     10.200.0.1       192.168.1.201:58517     75.88.115.189      [10.200.0.1/32]     false          
3926a387-fe28-43d0-97ab-c554fdcc5059     10.200.0.2       192.168.1.202:54762     75.88.115.189      [10.200.0.2/32]     false          


apexctl  --username=kitteh1@apex.local --password=floofykittens --output=json peer list --zone-id 32956ffd-3a6a-472e-82ad-a40b49b13b95
[
    {
        "id": "9a755246-dd41-4175-80a5-fb4179d7cc70",
        "device_id": "db629003-acde-4169-9620-61ab83c311f6",
        "zone_id": "32956ffd-3a6a-472e-82ad-a40b49b13b95",
        "endpoint_ip": "192.168.1.201:58517",
        "allowed_ips": [
            "10.200.0.1/32"
        ],
        "node_address": "10.200.0.1",
        "child_prefix": "",
        "hub_router": false,
        "hub_zone": false,
        "zone_prefix": "10.200.1.0/20",
        "reflexive_ip4": "75.88.115.189",
        "endpoint_local_address_ip4": "192.168.1.201",
        "symmetric_nat": false
    },
    {
        "id": "3926a387-fe28-43d0-97ab-c554fdcc5059",
        "device_id": "99f545a7-d695-4fdd-bc97-f1bd6014cf22",
        "zone_id": "32956ffd-3a6a-472e-82ad-a40b49b13b95",
        "endpoint_ip": "192.168.1.202:34388",
        "allowed_ips": [
            "10.200.0.2/32"
        ],
        "node_address": "10.200.0.2",
        "child_prefix": "",
        "hub_router": false,
        "hub_zone": false,
        "zone_prefix": "10.200.1.0/20",
        "reflexive_ip4": "75.88.115.189",
        "endpoint_local_address_ip4": "192.168.1.202",
        "symmetric_nat": false
    }
]

```

### List Devices

```
apexctl  --username=kitteh1@apex.local --password=floofykittens device list
DEVICE ID                                HOSTNAME         NODE ADDRESS     ENDPOINT IP          PUBLIC KEY                                       ZONE ID
078e8d95-72c0-41cd-bdd9-6b81f0ce8143     3f98ee1de2d9     172.19.100.1     172.17.0.5:55006     YO4b/oauALiOrAzSpo+wJhlEePZw4pfgp3cfLsdv4Rs=     0bbc1040-f1e4-4dfb-8f0c-4f7a3caf1f2c
5534b6e5-d506-4b50-888a-fa436ec7beaa     5b205061696f     172.19.100.2     172.17.0.4:36612     zhzUJvGkfYuCv1l8mgXXUZigBW43EPDVrnKHz0Az7xQ=     0bbc1040-f1e4-4dfb-8f0c-4f7a3caf1f2c
b0059e76-d8b3-4894-8fd8-c23f17327dd3     54e6d8412054     100.64.100.1     172.17.0.5:60501     SOWseq/GS+lCLdp8WchqcWLipOn95Cz9ML68j7A0LWY=     4e8f5374-9f73-41aa-b1fd-dfc7ae7cbc5f
fc796c50-4391-4605-8d46-00ca45803e86     5c2b07f8dc7a     100.64.100.2     172.17.0.4:42640     eM4eG8PlObYOfGvUz8Eobx4/QxHkIOcaFKKm1hmPyGk=     4e8f5374-9f73-41aa-b1fd-dfc7ae7cbc5f
6f8310b7-4d02-41bb-8d3f-c3f397e49e39     a787d93f0512     10.162.0.1       172.17.0.4:51820     1+ZFonrwTxDf+PpM5r0CrDMK7gmBcprG73MeR0L582c=     1983589e-e5d9-4de0-8410-a05300dceef2
0836ad93-cc0c-43bb-941c-80087d044a7f     a153291e2de8     10.162.0.2       172.17.0.6:40594     wee3XptO1viN/n2lSgBBCCGvddtZ8qVOA6kbpCbJZmY=     1983589e-e5d9-4de0-8410-a05300dceef2
e443a37e-11d5-432c-8b3e-cd507a8a72ad     f51e324afe1a     10.162.0.3       172.17.0.5:42982     9sHO1Vm5V4UWKdCrgfm2ra2sX0IlDgeVqsCjGwULMRU=     1983589e-e5d9-4de0-8410-a05300dceef2
119dd397-2abf-4baa-a8e1-285c435d67ff     0bb71f1172b9     10.200.0.1       172.17.0.5:59198     Q1OfpI4DUD9jR4xsZ7fGq+um0MVGCfp3m0PJ9mNA/18=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
8da7e36d-d519-4e61-990d-a2211b1c833e     e5e7356a5432     10.200.0.2       172.17.0.4:51487     YjMOI7txs4p9kdOE9ihnTN0PkdfG3w/1wxqzYMI7DEU=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
edfa55de-8905-416d-a168-076c73e3c687     0bb71f1172b9     10.200.0.3       172.17.0.5:58625     u5nK9s4HchMMO0egHCrkg2jigLWPoq1+O9tI0TVJhHE=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
5c7d0fcc-d56f-4b99-b649-4586a1ff2939     e5e7356a5432     10.200.0.4       172.17.0.4:33085     0CDvGj008P9cgXhC9XyZcuzmRB0pCGGhukdXigpFMzQ=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
e22e870b-5dfe-404d-b241-6bdba7caaec9     27f3b936c448     10.200.0.102     172.17.0.5:41657     iDUPNn8UDSlbF7mgstiNrnw2KWiVsxnFgKC8Hp8lol8=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
a032e773-fb64-4854-aefc-2614314d3ba0     132f3a264302     10.200.0.101     172.17.0.4:43773     0PLvrsD1I/3JGFsOFylCoqYNXlfjE4anhz6OpX09KzU=     89e7abd1-4627-40f8-9daf-6fcd41ff39c9
e0090168-db50-4f0e-94e2-704fb2cba089     6fb1f90c4b8d     10.140.0.102     172.17.0.5:58025     gUCBEjAEI8l04aGW4TKR8jO9SpHB3HYtjniT7aMLSBE=     2978ce28-287b-49fe-af1c-c455f802ee4b
515f2f47-89e5-4ada-88fa-7ae9447fa2a6     e3f75108f51c     10.140.0.101     172.17.0.4:42147     eCe2axzp5C/U4ghyYpISLLpMDYCfNy4o766j1H2Wq0I=     2978ce28-287b-49fe-af1c-c455f802ee4b

```


### Delete Device ###

```
apexctl  --username=kitteh1@apex.local --password=floofykittens device list
DEVICE ID                                HOSTNAME     NODE ADDRESS     ENDPOINT IP             PUBLIC KEY                                       ZONE ID
db629003-acde-4169-9620-61ab83c311f6     ub201        10.200.0.1       192.168.1.201:58517     CpJ6vyh8itq2H7lDd9E/MA9wM5TY72rkjpoPQc0FgAg=     32956ffd-3a6a-472e-82ad-a40b49b13b95
99f545a7-d695-4fdd-bc97-f1bd6014cf22     ub202        10.200.0.2       192.168.1.202:34388     dncfj+BG9AzHuGov+rNg6Xv/gid9jKfmLLxWaF19TSg=     32956ffd-3a6a-472e-82ad-a40b49b13b95

/apexctl  --username=kitteh1@apex.local --password=floofykittens device delete --device-id=99f545a7-d695-4fdd-bc97-f1bd6014cf22
successfully deleted device 99f545a7-d695-4fdd-bc97-f1bd6014cf22


apexctl  --username=kitteh1@apex.local --password=floofykittens device list
DEVICE ID                                HOSTNAME     NODE ADDRESS     ENDPOINT IP             PUBLIC KEY                                       ZONE ID
db629003-acde-4169-9620-61ab83c311f6     ub201        10.200.0.1       192.168.1.201:58517     CpJ6vyh8itq2H7lDd9E/MA9wM5TY72rkjpoPQc0FgAg=     32956ffd-3a6a-472e-82ad-a40b49b13b95

```

Reply:
It could be something from whatever is writing to stdout/stderr also - maybe a special character used to clear the terminal or something....
Either way, it's probably safe for containerExec to call `strings.Trim()` on the output - maybe even get rid of anything non UTF8 also.

Reply:
I tried all the trims and rune tricks. Using the binary from the apex/dist instead. Easier to read anyways.